src/main/java/com/epam/aidial/core/util/ProxyUtil.java

```diff
@@ -102,7 +106,11 @@ public static void collectAttachedFiles(ObjectNode tree, ApiKeyData apiKeyData)
                 JsonNode attachment = attachments.get(j);
                 JsonNode url = attachment.get("url");
                 if (url != null) {
-                    apiKeyData.getAttachedFiles().add(url.textValue());
+                    if (checkAccessFn.apply(url.textValue())) {
```

Comment:
this would not work, try to insert a valid public link and see how it's work

http: considered as wrong bucket and lead to access - false, which leads to 403 - but should be 200

Reply:
fixed